src/NameGenerator.elm

```diff
@@ -0,0 +1,116 @@
+module NameGenerator exposing
+    ( Definition, Lexicon
+    , generator
+    , capitalize
+    , fromString
+    )
+
+
+import Dict exposing (Dict)
+import Random.Pcg as Random exposing (Generator)
+import String
+import Regex as R
+
+
+
+-- types
+
+type Fragment
+    = Constant String
+    | Key String
+
+
+type alias Definition =
+    List Fragment
+
+
+type alias Lexicon =
+    Dict String (List Definition)
+
+
+
+-- Generate a name given a lexicon and a key of that lexicon
+
+generator : Lexicon -> String -> Generator String
+generator lexicon key =
+
+    case Dict.get key lexicon of
+
+        Nothing ->
+            -- either the key is plain invalid, or it is stuck in a loop
+            Random.constant <| "--[" ++ key ++ "]--"
```

Comment:
Try again to use `generator : Lexicon -> String -> Generator (Result (List String) String)`, and make error results chainable?